### DISCUSSION TITLE
GCN on 3d point  cloud data

### DISCUSSION THREAD

**Discussion by utkarsh0902311047** [pyg-team/pytorch_geometric#8053] [Q&A]
Hi,
I would like to ask :
1. if any code is available to implement GCN on 3D point cloud data.
2. Also, which is the example tutorial for point-gnn, I cannot find it.
3. Does Point Net use any Graph neural networks?

Thank you
Regards

**Comment by rusty1s** [comment]
You can search for `point*` in our examples folder, e.g., here is an example of `PointNet++` for classification.

**Reply by utkarsh0902311047** [reply]
I could only find pointnet2_classification and segmentation in the examples folder.
But I am looking for the example that uses torch_geometric/nn/conv/ point_gnn_conv.py for point-gnn implementation.

**Reply by rusty1s** [reply]
I see. We don't have an example for this, but you should be able to just switch out the `PointNetConv` operators with `PointGNNConv` in the given example. Does that work for you?

**Reply by utkarsh0902311047** [reply]
For PointGNNConv, PyG 2.3.0 is required. But when I use this PyG version I get error "TypeError: _new_shared() got an unexpected keyword argument 'device'." while training.
What should I do?
I am using cuda10.1, torch1.8.1

**Reply by utkarsh0902311047** [reply]
Please help

**Reply by rusty1s** [reply]
PyG 2.3 is only compatible for Pytorch >= 1.11.0. I think you may need to upgrade your PyTorch version, sorry :(

**Reply by utkarsh0902311047** [reply]
So I need a higher cuda version too I believe?

**Reply by rusty1s** [reply]
Yes, that's correct.

**Reply by utkarsh0902311047** [reply]
@rusty1s 
Hi
I tried running Point-GNN example by replacing PointNetConv ---> PointGNNConv.
But I am getting following error.
Traceback (most recent call last):
  File "C:\Users\utkarsh\Documents\pytorch_geometric\examples\pointGNN_classification.py", line 104, in <module>
    model = Net().to(device)
  File "C:\Users\utkarsh\Documents\pytorch_geometric\examples\pointGNN_classification.py", line 52, in __init__
    self.sa1_module = SAModule(0.5, 0.2, MLP([3, 64, 64, 128]))
  File "C:\Users\utkarsh\Documents\pytorch_geometric\examples\pointGNN_classification.py", line 21, in __init__
    self.conv = PointGNNConv(nn, add_self_loops=False)
TypeError: __init__() missing 2 required positional arguments: 'mlp_f' and 'mlp_g'

What changes should I make?
Please help

**Reply by rusty1s** [reply]
Here is a small example on how to use `PointGNNConv`:
```python
    conv = PointGNNConv(
        mlp_h=MLP([8, 16, 3]),
        mlp_f=MLP([3 + 8, 16, 8]),
        mlp_g=MLP([8, 16, 8]),
    )
```
so you see that it requires 2 additional MLP definitions. Hope this helps.
